src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/controller/ListingController.java

```diff
@@ -48,30 +44,169 @@ public class ListingController {
+                } catch (IllegalArgumentException e) {
+                    throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Gender for AVAILABLE_to does not exist");
+                }
+            }
+            filtersMap.put("AVAILABLE_TO", temp);
+        }
+        if (listingType.isPresent()) {
+            // for conversion from String to ENUM
+            List<ListingType> temp = new ArrayList<>();
+            for (String listingString : listingType.get()) {
+                try {
+                    temp.add(ListingType.valueOf(listingString.toUpperCase()));
+                } catch (IllegalArgumentException e) {
+                    throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "LISTING_TYPE does not exist");
+                }
+            }
+            filtersMap.put("LISTING_TYPE", temp);
+        }
+        if (city.isPresent()) filtersMap.put("CITY", city.get());
+        if (zipCode.isPresent()) filtersMap.put("ZIP_CODE", zipCode.get());
+        if (minSqm.isPresent()) {
+            try {
+                filtersMap.put("MIN_SQM", Integer.parseInt(minSqm.get()));
+            } catch (NumberFormatException e) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Expected a number");
+            }
+        }
+        if (maxSqm.isPresent()) {
+            try {
+                filtersMap.put("MAX_SQM", Integer.parseInt(maxSqm.get()));
+            } catch (NumberFormatException e) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Expected a number");
+            }
+        }
+
+        if (available.isPresent()) filtersMap.put("AVAILABLE", Boolean.parseBoolean(available.get()));
+
+        if (filtersMap.size() > 0) {
+            List<FilterPair> filterPairs = filtersMap.entrySet().stream()
+                    .map(e -> new FilterPair(ListingFilters.getFilter(e.getKey()), e.getValue())).toList();
+
+            allListings = ListingFilter.createFilter(allListings).filter(filterPairs).getListings();
+        }
+
+        if (sort.isPresent()) {
+
+            Sorting sorting;
+            try {
+                sorting = Sorting.valueOf(sort.get().toUpperCase());
+            } catch (IllegalArgumentException e) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Sorting not supported");
+            }
+
+            OrderType orderType = OrderType.ASC;
+
+            if (order.isPresent()) {
+                try {
+                    orderType = OrderType.valueOf(order.get().toUpperCase());
+                } catch (IllegalArgumentException e) {
+                    throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "OrderType not found, only ASC and DESC supported");
+                }
+            }
+
+            // set default ordering to ASC or use userinput, only works for rent at the moment
+            if (orderType.equals(OrderType.ASC)) {
```

Comment:
always sort `ASC` and do `Collections.reverse(LIST);` if `DESC`

Reply:
is this not possible or did you resolve this by accident?

Reply:
didn't work for me, so i made my custom reversing.... If you know how, please change it.